Calculator/NodeTests/NodeTests.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  NodeTests.swift
+//  NodeTests
+//
+//  Created by Daehoon Lee on 2023/05/30.
+//
+
+import XCTest
+@testable import Calculator
+
+final class NodeTests: XCTestCase {
+    var sut: Node<Int>!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        sut = Node<Int>(data: 0, next: nil)
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+        sut = nil
+    }
+    
+    func test_init_data를0으로초기화한다() {
+        // given
+        let expectedData: Int = 0
+        
+        // when
+        let testNode: Node = Node<Int>(data: expectedData, next: nil)
+        
+        // then
+        XCTAssertEqual(testNode.fetchData(), expectedData)
+    }
```

Comment:
테스트 케이스의 네이밍도 given / when / then의 흐름으로 작성하면 테스트를 파악하기 더 좋을 것 같습니다.

```suggestion
    func test_0으로초기화한노드의_fetchData를호출하면_0을반환한다() {
        // given
        sut = Node<Int>(data: 0, next: nil)
        
        // when
        let result = sut?.fetchData()
        
        // then
        XCTAssertEqual(result, 0)
    }
```

또한 `testNode`와 같은 지역변수에 대한 테스트를 피하는 것이 좋을 것 같습니다. 지역변수에 뜬금없는 타입을 갖다 놓고 테스트를 진행하더라도 문제없이 테스트 메서드가 동작할텐데 그 경우 잘못을 인지하지 못할 수도 있고,  테스트 클래스별로 역할이 분명할텐데 지역변수에 대한 테스트를 하는 순간부터 역할과 무관해질 수도 있을 것 같습니다.

Reply:
- 테스트 케이스의 네이밍을 말씀해주신 내용에 맞게 수정해보았습니다.🙇
- 제너릭을 사용할 때 `sut`에 `Int` 타입으로 선언하여 생성하였지만 테스트 케이스에서 `String` 타입을 사용하고 싶다면 이 때는 어떻게 처리를 할 수 있을까요? 이런 경우에도 지역변수에 대한 테스트를 피하는 것이 좋을까요?
- 테스트 케이스를 작성할 때 `given`에 맞는 조건이 없을 수도 있을까요?

Reply:
`given`에 맞는 조건이 없을 수는 없다고 생각합니다. 가령 비어있다거나 초기화를 했다거나 등등의 상황 설정이죠.

다른 타입으로 바꾸고 싶은 이유가 있을까요? 코드에 대한 검증은 타입과 무관할 것 같습니다.